build-common/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+use cbindgen::Config;
+use std::path::PathBuf;
+use std::process::Command;
+use std::str;
+use std::{env, fs};
+
+/// Configure the header generation using environment variables.
+/// call into `generate_header` with the appropriate arguments.
+///
```

Comment:
I can still find ways to break this, but I feel we already improve the state of the build.